UM/Settings/ContainerStack.py

```diff
@@ -194,10 +194,14 @@ def setDirty(self, dirty: bool) -> None:
     #   Note that if the property value is a function, this method will return the
     #   result of evaluating that property with the current stack. If you need the
     #   actual function, use getRawProperty()
-    def getProperty(self, key: str, property_name: str):
+    def getProperty(self, key: str, property_name: str, context: Optional[PropertyEvaluationContext] = None):
         value = self.getRawProperty(key, property_name)
         if isinstance(value, SettingFunction):
-            return value(self)
+            if context is None:
+                context = PropertyEvaluationContext(self)
```

Comment:
You don't need to create a PropertyEvaluationContext here because the `__call__` of `SettingFunction` accepts `None` as context. It would save some time, as this function is very time-critical.

Reply:
Fixed in 8a93ee003